### PR TITLE
Make sure that there are only ever 12 slots in a hotbar

### LINKED ISSUE #3255: on 12 clusters listed in hotbar
**Describe the bug**
adding a 13 cluster to the hotbar will do nothing except show a greyed out box as tho it was waiting for the user to add a cluster

**To Reproduce**

**Expected behavior**
clusters keep getting added. maybe the pagination works even

**Screenshots**
Screen Shot 2021-07-02 at 12 45 03 PM

**Environment (please complete the following information):**
- Lens Version: 5.0
- OS: [e.g. OSX] osx 10.15.7
- Installation method (e.g. snap or AppImage in Linux): brew

**Logs:**
When you run the application executable from command line you will see some logging output. Please paste them here:
```
Your logs go here...
```

**Kubeconfig:**
Quite often the problems are caused by malformed kubeconfig which the application tries to load. Please share your kubeconfig, remember to remove any secret and sensitive information.
```
your kubeconfig here
```

**Additional context**

### PR DESCRIPTION
**Pull request by Nokel81**
- Send notification if hotbar is full

- Filter out on every fromStore call

Signed-off-by: Sebastian Malton <sebastian@malton.name>

fixes #3255 

Screen Shot 2021-07-14 at 12 15 21 PM

### CODE REVIEW

File: src/common/hotbar-store.ts [hunk 1]
```diff
@@ -93,29 +94,6 @@ export class HotbarStore extends BaseStore<HotbarStoreModel> {
     return [...Array.from(Array(defaultHotbarCells).fill(null))];
   }
 
-  @action
```
Comment (Nokel81) [inline src/common/hotbar-store.ts]: I moved this down so that it would be closer to `toJSON`

File: src/common/hotbar-store.ts [hunk 2]
```diff
@@ -175,18 +153,24 @@ export class HotbarStore extends BaseStore<HotbarStoreModel> {
...
-      const emptyCellIndex = hotbar.items.findIndex(isNull);
+      const emptyCellIndex = hotbar.items.indexOf(null);
 
       if (emptyCellIndex != -1) {
         hotbar.items[emptyCellIndex] = newItem;
       } else {
-        // Add new item to the end of list
-        hotbar.items.push(newItem);
+        if (ipcRenderer) {
+          import("../renderer/components/notifications")
```
Comment (Nokel81) [inline src/common/hotbar-store.ts]: This is needed because it doesn't exist on `main`. I really wish we would split these into main and renderer sides.

Comment (jakolehm) [reply src/common/hotbar-store.ts]: I think we should just broadcast a message and act on that within renderer. Would keep store cleaner.

Comment (Nokel81) [reply src/common/hotbar-store.ts]: Okay, it just seems like hotbar-store is really a renderer-only thing but I can change it.

File: src/common/hotbar-store.ts [hunk 3]
```diff
@@ -175,18 +153,19 @@ export class HotbarStore extends BaseStore<HotbarStoreModel> {
...
 
       if (emptyCellIndex != -1) {
         hotbar.items[emptyCellIndex] = newItem;
       } else {
-        // Add new item to the end of list
-        hotbar.items.push(newItem);
+        broadcastMessage(HotbarTooManyItems);
       }
-    } else {
+    } else if (0 <= cellIndex && cellIndex <= hotbar.items.length) {
```
Comment (jim-docker) [inline src/common/hotbar-store.ts]: Shouldn't it be
```suggestion
    } else if (0 <= cellIndex && cellIndex < hotbar.items.length) {
```

Comment (Nokel81) [reply src/common/hotbar-store.ts]: yes it should

File: src/common/hotbar-store.ts [hunk 4]
```diff
@@ -175,18 +153,19 @@ export class HotbarStore extends BaseStore<HotbarStoreModel> {
...
 
       if (emptyCellIndex != -1) {
         hotbar.items[emptyCellIndex] = newItem;
       } else {
-        // Add new item to the end of list
-        hotbar.items.push(newItem);
+        broadcastMessage(HotbarTooManyItems);
       }
-    } else {
+    } else if (0 <= cellIndex && cellIndex < hotbar.items.length) {
```
Comment (jakolehm) [inline src/common/hotbar-store.ts:187]: Would be nice to see tests for this.

**Comment by chenhunghan** [comment]
On this branch, I got this when trying to pin `WebLink: Kubernetes Documentation` to hotbar
Screenshot 2021-08-02 at 15 16 02

File: src/common/hotbar-store.ts [hunk 5]
```diff
@@ -163,39 +141,52 @@ export class HotbarStore extends BaseStore<HotbarStoreModel> {
...
+    if (typeof uid !== "string") {
+      throw new TypeError("CatalogEntity.getId() must return a string");
+    }
+
+    if (typeof name !== "string") {
+      throw new TypeError("CatalogEntity.getName() must return a string");
+    }
+
     const newItem = { entity: {
-      uid: item.metadata.uid,
```
Comment (aleksfront) [inline src/common/hotbar-store.ts:145]: Not all entities currently have `getItem()` method, so `safeCall` won't help here. And that's why @chenhunghan has an error when trying to pin some of the Links to hotbar.

Comment (aleksfront) [reply src/common/hotbar-store.ts:145]: Actually, **none** of the entities can be pinned to Hotbar. 🐞

**Review by chenhunghan — APPROVED** [review]
Tested locally and confirm the TypeError when adding items to catalog is gone.

**Pull request merged**